content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2422 @@
+
+In a similar spirit, there are situation where excluding every unpopulated
+bucket from the bucket list is most efficient, but in other situations, it
+might be better to reduce the number of spans by representing small numbers of
+unpopulated buckets explicitly.
+
+Note that future high resolution schemas might require offsets that are too
+large to be represented with an int32. An extension of the data model will be
+required in that case. (The current standard schema with the highest resolution
+is schema 8, for which the bucket that contains `MaxFloat64` has index 262144,
+and thus the `+Inf` overflow bucket has index 262145, while the largest number
+representable with int32 is 2147483647. The highest standard schema that would
+still work with int32 offsets would be schema 20, corresponding to a growth
+factor from bucket to bucket of only ~1.000000661.)
+
+#### Examples
+
+An integer histogram has the following positive buckets (index→population):
+
+`-2→3, -1→5, 0→0, 1→0, 2→1, 3→0, 4→3, 5→2`
+
+They could be represented in this way:
+
+- Positive bucket list: `[3, 2, -4, 2, -1]`
+- Positive span list: `[[-2, 2], [2,1], [1,2]]`
+
+The second and third span could be merged into one if the single unpopulated
+bucket with index 3 is represented explicitly, leading to the following result:
+
+- Positive bucket list: `[3, 2, -4, -1, 3, -1]`
+- Positive span list: `[[-2, 2], [2,4]]`
+
+Or merge all the spans into one by representing all unpopulated buckets above
+explicitly:
+
+- Positive bucket list: `[3, 2, -5, 0, 1, -1, 3, -1]`
+- Positive span list: `[[-2, 8]]`
+
+### Zero bucket
+
+Observations of exactly zero do not fit into any bucket as defined by the
+standard schemas above. They are counted in a dedicated bucket called the _zero
+bucket_.
+
+The number of observations in the zero bucket is tracked by a single uint64
+(for integer histograms) or float64 (for float histograms).
+
+The zero bucket has an additional parameter called the _zero threshold_, which
+is a float64 ≥ 0. If the threshold is set to zero, only observations of exactly
+zero go into the zero bucket, which is the case described above. If the
+threshold has a positive value, all observations within the closed interval
+[-threshold, +threshold] go to the zero bucket rather than a regular bucket.
+This has two use cases:
+
+- Noisy observations close to zero tend to populate a high number of buckets.
+  Those observations might happen due to numerical inaccuracies or if the
+  source of the observations are actual physical measurements. A zero bucket
+  with a relatively small threshold redirects those observations into a single
+  bucket.
+- If the user is more interested in the long tail of a distribution, far away
+  from zero, a relatively large threshold of the zero bucket helps to avoid
+  many high resolution buckets for a range that is not of interest.
+
+The threshold of the zero bucket SHOULD coincide with a boundary of a regular
```

Comment:
Do this should be "SHOULD NOT" instead of "SHOULD"? I take "coincide with" as "same or overlapped".